frontend/src/libs/components/badge/badge.tsx

```diff
@@ -0,0 +1,49 @@
+import { getValidClassNames } from '~/libs/helpers/helpers.js';
+
+import darkColors from './dark-colors.module.scss';
+import lightColors from './light-colors.module.scss';
+import styles from './styles.module.scss';
+import { type DarkColor, type LightColor } from './types/color.type.js';
+
+type Color = DarkColor | LightColor;
+
+const paletteColors = {
+  ...(darkColors as { [K in DarkColor]: string }),
+  ...(lightColors as { [K in LightColor]: string }),
+};
+const [defaultColor] = Object.keys(paletteColors) as Color[];
+
+type Properties = {
+  children: string;
+  className?: string;
+  color?: Color;
+  onClick?: (event: React.MouseEvent<HTMLButtonElement>) => void;
+};
+
+const Badge: React.FC<Properties> = ({
+  children,
+  className,
+  color = defaultColor,
+  onClick: handleClick,
```

Comment:
Not necessary to rename handler here. We should name it with `handle` prefix only during declaration.

Reply:
fixed